elephant/statistics.py

```diff
@@ -974,23 +977,414 @@ def complexity_pdf(spiketrains, bin_size):
-
-    # Computing the histogram of the entries of pophist (=Complexity histogram)
-    complexity_hist = np.histogram(
-        pophist.magnitude, bins=range(0, len(spiketrains) + 2))[0]
-
-    # Normalization of the Complexity Histogram to 1 (probabilty distribution)
-    complexity_hist = complexity_hist / complexity_hist.sum()
-    # Convert the Complexity pdf to an neo.AnalogSignal
-    complexity_distribution = neo.AnalogSignal(
-        np.expand_dims(complexity_hist, axis=1) *
-        pq.dimensionless, t_start=0 * pq.dimensionless,
-        sampling_period=1 * pq.dimensionless)
-
-    return complexity_distribution
+    warnings.warn("complexity_pdf is deprecated in favor of the complexity "
+                  "class which has a pdf attribute. complexity_pdf will be "
+                  "removed in the next Elephant release.", DeprecationWarning)
+
+    complexity = Complexity(spiketrains, bin_size=bin_size)
+
+    return complexity.pdf()
+
+
+class Complexity(object):
+    """
+    Class for complexity distribution (i.e. number of synchronous spikes found)
+    of a list of `neo.SpikeTrain` objects.
+
+    Complexity is calculated by counting the number of spikes (i.e. non-empty
+    bins) that occur separated by `spread - 1` or less empty bins, within and
+    across spike trains in the `spiketrains` list.
+
+    Implementation (without spread) is based on [1]_.
+
+    Parameters
+    ----------
+    spiketrains : list of neo.SpikeTrain
+        Spike trains with a common time axis (same `t_start` and `t_stop`)
+    sampling_rate : pq.Quantity or None, optional
+        Sampling rate of the spike trains with units of 1/time.
+        Default: None
+    bin_size : pq.Quantity or None, optional
+        Width of the histogram's time bins with units of time.
+        The user must specify the `bin_size` or the `sampling_rate`.
+          * If no `bin_size` is specified and the `sampling_rate` is available
+            1/`sampling_rate` is used.
+          * If both are given then `bin_size` is used.
+        Default: None
+    binary : bool, optional
+          * If `True` then the time histograms will be binary.
+          * If `False` the total number of synchronous spikes is counted in the
+            time histogram.
+        Default: True
+    spread : int, optional
+        Number of bins in which to check for synchronous spikes.
+        Spikes that occur separated by `spread - 1` or less empty bins are
+        considered synchronous.
+          * ``spread = 0`` corresponds to a bincount accross spike trains.
+          * ``spread = 1`` corresponds to counting consecutive spikes.
+          * ``spread = 2`` corresponds to counting consecutive spikes and
+            spikes separated by exactly 1 empty bin.
+          * ``spread = n`` corresponds to counting spikes separated by exactly
+            or less than `n - 1` empty bins.
+        Default: 0
+    tolerance : float, optional
```

Comment:
If None can be passed, the information and behavior should be added here. If warnings are raised, then a section Warns can be added

Reply:
Done